File: server/loans/core/src/main/java/ee/rsx/kata/bank/loans/domain/segment/CreditSegment.java
package ee.rsx.kata.bank.loans.domain.segment;

import ee.rsx.kata.bank.loans.domain.ssn.SocialSecurityNumber;

import static ee.rsx.kata.bank.loans.domain.segment.CreditSegmentType.DEBT;

public record CreditSegment(
  SocialSecurityNumber ssn,
  CreditSegmentType type,
  int creditModifier
) {
 public int creditModifier() {

   return type == DEBT ? 0 : this.creditModifier;
 }
 public boolean isDebt() {
   return type == DEBT;
 }
}

File: server/loans/core/src/main/kotlin/ee/rsx/kata/bank/loans/domain/segment/CreditSegment.kt
package ee.rsx.kata.bank.loans.domain.segment

import ee.rsx.kata.bank.loans.domain.segment.CreditSegmentType.DEBT
import ee.rsx.kata.bank.loans.domain.ssn.SocialSecurityNumber

data class CreditSegment(
  val ssn: SocialSecurityNumber,
  val type: CreditSegmentType,
  private val modifier: Int
) {
  val isDebt
    get() = type === DEBT

  val creditModifier
    get() = if (isDebt) 0 else modifier
}
